.github/workflows/ci.yml

```
@@ -1,66 +1,209 @@
       - name: Install dependencies
         run: npm ci
-      - name: Run tests
+
+      - name: Run unit tests
         run: npm run test:run
 
+      - name: Generate test coverage
+        run: npm run test:coverage
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.9)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)